packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -281,247 +285,85 @@ function contextualizeInsertedArrayContent(
 // to the array node proxy.  Over time, we should replace these with efficient implementations on LazySequence
 // to avoid re-entering the proxy as these methods access 'length' and the indexed properties.
 //
 // For brevity, the current implementation dynamically builds a property descriptor map from a list of
 // Array functions we want to re-expose via the proxy.
 
-// TODO: This assumes 'Function.name' matches the property name on 'Array.prototype', which may be
-// dubious across JS engines.
-[
-	Array.prototype.concat,
-	// Array.prototype.copyWithin,
-	Array.prototype.entries,
-	Array.prototype.every,
-	// Array.prototype.fill,
-	Array.prototype.filter,
-	Array.prototype.find,
-	Array.prototype.findIndex,
-	Array.prototype.flat,
-	Array.prototype.flatMap,
-	Array.prototype.forEach,
-	Array.prototype.includes,
-	Array.prototype.indexOf,
-	Array.prototype.join,
-	Array.prototype.keys,
-	Array.prototype.lastIndexOf,
-	// Array.prototype.length,
-	Array.prototype.map,
-	// Array.prototype.pop,
-	// Array.prototype.push,
-	Array.prototype.reduce,
-	Array.prototype.reduceRight,
-	// Array.prototype.reverse,
-	// Array.prototype.shift,
-	Array.prototype.slice,
-	Array.prototype.some,
-	// Array.prototype.sort,
-	// Array.prototype.splice,
-	Array.prototype.toLocaleString,
-	Array.prototype.toString,
-	// Array.prototype.unshift,
-	Array.prototype.values,
-].forEach((fn) => {
-	arrayNodePrototypeProperties[fn.name] = { value: fn };
+const arrayPrototypeKeys = [
+	"concat",
+	"entries",
+	"every",
+	"filter",
+	"find",
+	"findIndex",
+	"flat",
+	"flatMap",
+	"forEach",
+	"includes",
+	"indexOf",
+	"join",
+	"keys",
+	"lastIndexOf",
+	"map",
+	"reduce",
+	"reduceRight",
+	"slice",
+	"some",
+	"toLocaleString",
+	"toString",
+	"values",
+
+	// "copyWithin",
+	// "fill",
+	// "length",
+	// "pop",
+	// "push",
+	// "reverse",
+	// "shift",
+	// "sort",
+	// "splice",
+	// "unshift",
+] as const;
+
+/**
+ * {@link TreeNodeValid}, but modified to add members from Array.prototype named in {@link arrayPrototypeKeys}.
+ */
+abstract class TreeNodeWithArrayFeaturesUntyped<
+	const T extends ImplicitAllowedTypes,
+> extends TreeNodeValid<Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>> {}
+
+arrayPrototypeKeys.forEach((key) => {
+	Object.defineProperty(TreeNodeWithArrayFeaturesUntyped.prototype, key, {
+		value: Array.prototype[key],
+	});
 });
 
-/* eslint-enable @typescript-eslint/unbound-method */
+/**
+ * Type of {@link TreeNodeWithArrayFeaturesUntyped}, but with its array members added to the type.
+ */
+type TreeNodeWithArrayFeaturesType = TreeNodeValueStatics &
+	(abstract new <T extends ImplicitAllowedTypes>(
+		input: Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>> | InternalTreeNode,
+	) => TreeNodeValid<Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>> &
+		Pick<readonly TreeNodeFromImplicitAllowedTypes<T>[], (typeof arrayPrototypeKeys)[number]>);
```

Comment:
This makes the inclusion of the members from array.prototype type safe. Doing this found a bug: unscopables was missing. Pretty insignificant bug, but much better type safety.